network/gateway.go

```diff
@@ -0,0 +1,95 @@
+package network
+
+import (
+	"net"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+var simulatedOS = ""
+
+// GetDefaultRoute returns the IP address and device name of default gateway on the machine.
+// If we don't support OS, or we don't have a default route, we return nothing (not an error!).
+func GetDefaultRoute() (net.IP, string, error) {
+	os := simulatedOS
+	if os == "" {
+		os = runtime.GOOS
+	}
+	// TODO(wpk) 2017-11-20 Add Windows support here, hence the switch.
+	switch os {
+	case "linux":
+		return getDefaultRouteLinux()
+	default:
+		return nil, "", nil
+	}
+}
+
+func parseIpRouteShowLine(line string) (string, map[string]string) {
+	values := make(map[string]string)
+	fields := strings.Fields(line)
+	if len(fields) < 2 {
+		return "", values
+	}
+	to, fields := fields[0], fields[1:]
+	for ; len(fields) >= 2; fields = fields[2:] {
+		values[fields[0]] = fields[1]
+	}
+	return to, values
+}
+
+func launchIpRouteShowReal() (string, error) {
+	output, err := exec.Command("ip", "route", "show").CombinedOutput()
```

Comment:
Would it be better to use something like:
https://godoc.org/golang.org/x/net/route
We already depend on golang.org/x/net and that would let us avoid spawning and parsing in order to get route information.

Reply:
x/net/route is for BSD only, I was looking for a library to do it and have found nothing (well, nothing maintained at least).